docs/performance/device-tensor.md

```diff
@@ -0,0 +1,136 @@
+
+### Python API
+
+The Python API supports the same performance opportunities as the above mentioned C++ API.
+[Device tensors](https://onnxruntime.ai/docs/api/python/api_summary.html#data-on-device) can be allocated as shown here.
+Besides this the `user_compute_stream` can be set through this [API](https://onnxruntime.ai/docs/api/python/api_summary.html#onnxruntime.InferenceSession.set_providers)
+
+```python
+sess = onnxruntime.InferenceSession("model.onnx", providers=["TensorrtExecutionProvider"])
+option = {}
+s = torch.cuda.Stream()
+option["user_compute_stream"] = str(s.cuda_stream)                    
+sess.set_providers(["TensorrtExecutionProvider"], [option])
+```
+
+Enabling asynchronous execution in python is possible through the same [run option](https://onnxruntime.ai/docs/api/python/api_summary.html#runoptions) as on the C++ API.
+
+## DirectML
+
+Acheiving the same behaviour is possible through DirectX resources.
+To run asynchronous processing it is crucial to do the same manamgemnt of execution streams as needed with CUDA.
+For DirectX this means managing the device and it's command queue which is possible through the C API.
+Details of how to set the compute command queue are documented with the usaged of [`SessionOptionsAppendExecutionProvider_DML1`](../execution-providers/DirectML-ExecutionProvider.md#usage).
+
+If separate command queues are used for copy and compute it is possible to overlap PCI copies and execution as well as make execution asynchronous.
+
+```c++
+#include <onnxruntime/dml_provider_factory.h>
+Ort::MemoryInfo memory_info_dml("DML", OrtDeviceAllocator, device_id,
+                                OrtMemTypeDefault);
+
+std::array<int64_t, 4> shape{1, 4, 64, 64};
+void *dml_resource;
+size_t d3d_buffer_size = 4 * 64 * 64 * sizeof(float);
+const OrtDmlApi *ort_dml_api;
+Ort::ThrowOnError(Ort::GetApi().GetExecutionProviderApi(
+                  "DML", ORT_API_VERSION, reinterpret_cast<const void **>(&ort_dml_api)));
+
+// Create d3d_buffer using D3D12 APIs
+Microsoft::WRL::ComPtr<ID3D12Resource> d3d_buffer = ...;
+
+// Create the dml resource from the D3D resource.
+ort_dml_api->CreateGPUAllocationFromD3DResource(d3d_buffer.Get(), &dml_resource);
+
+
+Ort::Value ort_value(Ort::Value::CreateTensor(memory_info_dml, dml_resource,
+                     d3d_buffer_size, shape.data(), shape.size(),
+                     ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT));
+```
+
+A [single file sample](https://github.com/ankan-ban/HelloOrtDml/blob/main/Main.cpp) can be found on github which shows how to manage and create copy and execution command queues.
+
+### Python API
+
+Although allocating DirectX inputs from python might not be a big use case the API is available as well. For intermediate network caches e.g. KV caching in LLMs this can be very beneficial.
```

Comment:
```suggestion

ChatGPT
Although allocating DirectX inputs from Python might not be a major use case, the API is available. This can prove to be very beneficial, especially for intermediate network caches, such as key-value caching in large language models (LLMs).
```